Civi/Core/CiviEventDispatcher.php

```diff
@@ -15,7 +15,7 @@
  *
  * @see \CRM_Utils_Hook
  */
-class CiviEventDispatcher extends ContainerAwareEventDispatcher {
+class CiviEventDispatcher extends EventDispatcher {
```

Comment:
Aah, this is a BC break that removes some functions - eg `getContainer()`, `addListenerService()`, `addSubscriberService()`.

Personally, losing `getContainer()` is fine. (If someone really needed a container reference, they'd be more likely to use `Civi::container()` than `Civi::disptacher()->getContainer()`...) But losing `addListenerService()` feels a bit more dephell-y. Both flexmailer+mosaico use `addListenerService()`.

I suspect that anyone on D7/WP/J + Mosaico would be looking at WSOD under normal upgrade procedures:

* If you upgrade `civicrm-core` first, then you'd get WSOD (i.e. because `addListenerService()` is used by the older/pre-upgrade ext) - so you don't get a chance to update flexmailer+mosaico.
* If you upgrade flexmailer+mosaico first, then you'd get WSOD (i.e. because `ServiceClosureArgument` isn't a thing in Symfony 2.8) before you can update civicrm-core.

Perhaps we can just re-create `addListenerService()` directly in `CiviEventDispatcher` (since we can't inherit it)? Then it doesn't matter how you mix/match the versions - b/c `civicrm-core` supports the same contract either way.

I've pushed up a commit which does that. `flexmailer` (both current `master` and latest `v1.1.1`) passes its unit-tests run with this.

(In principle, re-creating `addSubscriberService()` would make sense too... but I don't think anyone in `universe` uses it. The code in `\Civi\FlexMailer\Services::registerListeners()` doesn't matter because that function never runs, and it hasn't since v0.1.-alpha1...)

Reply:
Yeh that is ok by me @totten